behaviour/graql/language/mutate.feature

```diff
@@ -0,0 +1,695 @@
+      when {
+        $p isa person, has name "robert";
+      }, then {
+        $p has nickname "bob";
+      };
+      """
+
+
+  ##################
+  # ABSTRACT TYPES #
+  ##################
+
+  Scenario: add abstract to existing entity type makes it abstract
+    Given graql define
+      """
+      define person abstract;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x sub person; $x abstract; get;
+      """
+    Then concept identifiers are
+      |     | check | value  |
+      | PER | label | person |
+    Then uniquely identify answer concepts
+      | x   |
+      | PER |
+
+
+  Scenario: add abstract to existing relation type makes it abstract
+    Given graql define
+      """
+      define employment abstract;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x sub employment; $x abstract; get;
+      """
+    Then concept identifiers are
+      |     | check | value      |
+      | EMP | label | employment |
+    Then uniquely identify answer concepts
+      | x   |
+      | EMP |
+
+
+  Scenario: add abstract to existing attribute type makes it abstract
+    Given graql define
+      """
+      define name abstract;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x sub name; $x abstract; get;
+      """
+    Then concept identifiers are
+      |     | check | value |
+      | NAM | label | name  |
+    Then uniquely identify answer concepts
+      | x   |
+      | NAM |
+
+
+  Scenario: add abstract to existing entity type throws on commit if it has an existing instance
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "Jeremy", has email "jeremy@grakn.ai";
+      """
+    Given the integrity is validated
+    Then graql define throws
+      """
+      define person abstract;
+      """
+
+
+  Scenario: add abstract to existing relation type throws on commit if it has an existing instance
+
+
+  Scenario: add abstract to existing attribute type throws on commit if it has an existing instance
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "Jeremy", has email "jeremy@grakn.ai";
+      """
+    Given the integrity is validated
+    Then graql define throws
+      """
+      define name abstract;
+      """
+
+
+  @ignore
+  # TODO: re-enable when concrete types cannot have abstract subtypes
+  Scenario: change concrete type to abstract throws on commit if it has a concrete supertype
+
+  @ignore
+  # TODO: re-enable when rules cannot infer abstract relations
+  Scenario: change concrete relation type to abstract throws on commit if it is the conclusion of any rule
+
+  @ignore
+  # TODO: check if rules can infer abstract attributes
+  Scenario: change concrete attribute type to abstract throws on commit if it is the conclusion of any rule
+
+  ###############
+  # INHERITANCE #
```

Comment:
In my mind, "inheritance" has to do with the fact that a subtype gets whatever the supertype has - do all these tests mutate hierarchies and then check that properties are preserved downards?

If not, maybe a better description is "Hierarchy Mutation"?

Reply:
Originally it was called "Schema Mutation: Inheritance", then I moved the tests to this file and dropped the prefix. I agree that Hierarchy Mutation is a better name.

Reply:
Renamed to Hierarchy Mutation.

Reply:
cool